packages/eslint-plugin-next/lib/rules/no-img-element.js

```diff
@@ -18,6 +20,11 @@ module.exports = {
         if (node.attributes.length === 0) {
           return
         }
+        
+        const attributes = new NodeAttributes(node)
+        if (attributes.has('src') && attributes.value('src').endsWith('.svg')) {
```

Comment:
This is looking at the file name instead of the content type so it wont work as expected for URLs that don't end in `.svg` but still serve SVG content.

Reply:
Thanks for catching that! but, the rule is for the case when ESLint can infer the type from the extension, I think you mean if the URL ends with `.svg` but the another content type if served with the correct `Content-Type` header.

In that case I agree, and this rule would not be able to catch that. Based on this I don't think it's a good idea to accept this change and I will continue to ignore this rule inline for cases when I am using SVG files.

Reply:
+1. Otherwise, I agree that it would be useful to have this check in place!